src/Illuminate/Foundation/Bootstrap/HandleExceptions.php

```diff
@@ -46,7 +48,11 @@ public function bootstrap(Application $app)
 
         set_exception_handler($this->forwardsTo('handleException'));
 
-        register_shutdown_function($this->forwardsTo('handleShutdown'));
+        if (! static::$globalShutdownFunctionRegistered) {
+            static::$globalShutdownFunctionRegistered = true;
+
+            register_shutdown_function($this->forwardsTo('handleShutdown'));
```

Comment:
Tbh I'd just rewrite the class to completely static implementation as I don't see any advantage of this now "hybrid" approach.

As that would break the API, I think it's fine to merge this as is since it affects only test suite and as stated in the original refactoring (https://github.com/laravel/framework/pull/40656#issue-1115353851): 

>In addition, this pull request offers an HandleExceptions::forgetApp that gets used by our own TestCase feature test class, to ensure the error handler set by the framework never gets used even if PHPUnit doesn't set its own error handler - it may happen on certain PHPUnit options.